hooks/firebird/use-event-stream.tsx

```diff
@@ -114,59 +122,39 @@ const useEventStream = (deploymentId: string, deploymentStatus?: string) => {
     [updateDeploymentProgress, updateDeploymentStatus],
   )
 
-  /**
-   * need to follow the appropriate event stream based on the instance action taking place (Start, Deploy, Stop)
-   * @returns method string for the event stream api call
-   */
-  const getMethod = (): string => {
-    switch (deploymentStatus) {
-      case DeploymentStatus.STARTING:
-        return "follow_deployment_start"
-      case DeploymentStatus.STOPPING:
-      case DeploymentStatus.STOPPED:
-        return "follow_deployment_stop"
-      case DeploymentStatus.DEPLOYING:
-        return "follow_deployment"
-      case DeploymentStatus.ACTIVE:
-        return lastDeploymentStatus === DeploymentStatus.STARTING
-          ? "follow_deployment_start"
-          : "follow_deployment"
-      default:
-        return "follow_deployment"
-    }
-  }
-
-  const method = getMethod()
+  const getMethod = useCallback(
+    (deploymentStatus: DeploymentStatus): string => {
+      switch (deploymentStatus) {
+        case DeploymentStatus.STARTING:
+          return "follow_deployment_start"
+        case DeploymentStatus.STOPPING:
+          return "follow_deployment_stop"
+        case DeploymentStatus.DEPLOYING:
+          return "follow_deployment"
+        case DeploymentStatus.PENDING:
+          return "follow_deployment"
+        default:
+          return "follow_deployment"
+      }
+    },
+    [],
+  )
 
   useEffect(() => {
+    // only open event stream if we are deploying, starting, or stopping
+    if (deploymentStatus === undefined || !eventStreamStatuses.includes(deploymentStatus)) {
+      return
+    }
+
     if (!sseUrl) {
       console.error("NEXT_PUBLIC_FIREBIRD_SSE_URL is not set")
       return
     }
-
-    console.log("some ref current", sseRef.current)
-  
     // Open EventSource only if there is an active action, and close it otherwise
     if (!sseRef.current) {
-      // To keep method consistent with the event stream and avoid unnecessary call errors
-      switch (deploymentStatus) {
-        case DeploymentStatus.STARTING:
-          setLastDeploymentStatus(deploymentStatus)
-          break
-
-        case DeploymentStatus.STOPPING:
-          // case DeploymentStatus.STOPPED:
-          setLastDeploymentStatus(DeploymentStatus.STOPPING)
-          break
-
-        case DeploymentStatus.DEPLOYING:
-          setLastDeploymentStatus(deploymentStatus)
-          break
-      }
-
       console.log("SSE: useEffect - creating new EventSource")
       sseRef.current = new EventSource(
-        `${sseUrl}?method=${method}&target=${deploymentId}`,
+        `${sseUrl}?method=${getMethod(deploymentStatus)}&target=${deploymentId}`,
```

Comment:
I guess it's technically "less code" calling getMethod here vs having an isolated function call and then inserting method as a template literal here...

Reply:
either works, i was just trying to get this done as an example